core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -69,7 +69,7 @@ public PartitionData copy() {
   /**
    * Used by Avro reflection to instantiate this class when reading manifest files.
    */
-  public GenericDataFile(org.apache.avro.Schema avroSchema) {
+  GenericDataFile(org.apache.avro.Schema avroSchema) {
```

Comment:
This needs to be public so that Avro can call it.

Reply:
Ah ok, we'll suppress the warnings here then since Checkstyle thinks the `public` modifier is redundant.